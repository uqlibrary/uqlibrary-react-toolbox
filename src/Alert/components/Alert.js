import React from 'react';
import {PropTypes} from 'prop-types';
import FontIcon from 'material-ui/FontIcon';
import FlatButton from 'material-ui/FlatButton';
import IconButton from 'material-ui/IconButton';
import NavigationClose from 'material-ui/svg-icons/navigation/close';

const Alert = ({title, message, type, action, actionButtonLabel, allowDismiss, dismissAction}) => {
    return (
        <div className="forAlerts">
            <div className={type + ' alertWrapper'}>
                <div className="columns is-multiline is-mobile">
                    <div className="column is-narrow alertIcon">
                        <FontIcon className="material-icons">{type}</FontIcon>
                    </div>
                    <div className="column alertText">
                        <div><b>{title}</b>&nbsp;-&nbsp;{message}</div>
                    </div>
                    {
                        action && actionButtonLabel &&
                        <div className="column is-narrow is-12-mobile">
                            <FlatButton
                                label={actionButtonLabel}
                                onTouchTap={action}
                                className="alertAction"/>
                        </div>
                    }
                    {
                        allowDismiss && dismissAction &&
                        <div className="column is-narrow is-hidden-mobile">
                            <IconButton onTouchTap={dismissAction}>
                                <NavigationClose className="alertDismiss"/>
                            </IconButton>
                        </div>
                    }
                </div>
            </div>
        </div>
    );
};

Alert.propTypes = {
    message: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    type: PropTypes.oneOf(['error', 'error_outline', 'warning', 'info', 'info_outline', 'help', 'help_outline', 'done']),
    action: PropTypes.func,
    actionButtonLabel: PropTypes.string,
    allowDismiss: PropTypes.bool,
    dismissAction: PropTypes.func,
};

Alert.defaultProps = {
    message: 'Unexpected error',
    title: 'Error',
    type: 'error',
    allowDismiss: false
};

export default Alert;
