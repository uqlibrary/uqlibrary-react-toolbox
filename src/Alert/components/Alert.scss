@import '../../../sass/variables';

// Desktop
@media (min-width: 768px) {
  .alertWrapper {
    border-radius: 3px;
    margin: 24px 0 !important;
    padding: 12px;
    min-height: 40px;

    .alertIcon {
      float: left;
      margin-right: -6px;
      height: 50px;
      span {
        font-size: 50px !important;
      }
    }
    .alertText {
      font-weight: bold;
      min-height: 50px !important;
      margin: 0px !important;
      padding: 12px !important;
      .alertTitle {
        font-weight: bold !important;
      }
    }
  }

  // Help
  &.help, &.help_outline {
    border: 1px solid lighten($theme-color, 10%);
    background-color: darken($theme-color, 5%);
    .alertText, .alertTitle {
      color: $text-color-reverse !important;
    }
    .alertIcon span {
      color: $text-color-reverse !important;
    }
  }

  // Error
  &.error, &.error_outline {
    border: 1px solid lighten($error-color, 10%);
    background-color: darken($error-color, 5%);
    .alertText, .alertTitle {
      color: $text-color-reverse !important;
    }
    .alertIcon span {
      color: $text-color-reverse !important;
    }
  }

  // Warning
  &.warning {
    border: 1px solid lighten($warning-color, 10%);
    background-color: darken($warning-color, 0%);
    .alertText, .alertTitle {
      color: $text-color-reverse !important;
    }
    .alertIcon span {
      color: $text-color-reverse !important;
    }
  }

  // Info
  &.info, &.info_outline {
    border: 1px solid lighten($info-color, 10%);
    background-color: darken($info-color, 5%);
    .alertText, .alertTitle {
      color: $text-color-reverse;
    }
    .alertIcon span {
      color: rgba($text-color-reverse, 0.9) !important;
    }
  }

}

// Mobile
@media (max-width: 768px) {
  .alertWrapper {
    border-radius: 3px;
    margin: 24px 0 !important;
    padding: 0 12px 24px 12px;
    min-height: 40px;

    .alertIcon {
      float: left;
      margin-bottom: -20px;
      padding: 2x 12px 12px 6px !important;
      span {
        font-size: 30px !important;
        line-height: 40px;
      }
    }
    .alertText {
      font-weight: bold;
    }

    // Help
    &.help, &.help_outline {
      border: 1px solid lighten($theme-color, 10%);
      background-color: darken($theme-color, 5%);
      .alertText, .alertTitle {
        color: $text-color-reverse !important;
      }
      .alertIcon span {
        color: $text-color-reverse !important;
      }
    }

    // Error
    &.error, &.error_outline {
      border: 1px solid lighten($error-color, 10%);
      background-color: darken($error-color, 5%);
      .alertText, .alertTitle {
        color: $text-color-reverse !important;
      }
      .alertIcon span {
        color: $text-color-reverse !important;
      }
    }

    // Warning
    &.warning {
      border: 1px solid lighten($warning-color, 10%);
      background-color: darken($warning-color, 0%);
      .alertText, .alertTitle {
        color: $text-color-reverse !important;
      }
      .alertIcon span {
        color: $text-color-reverse !important;
      }
    }

    // Info
    &.info, &.info_outline {
      border: 1px solid lighten($info-color, 10%);
      background-color: darken($info-color, 5%);
      .alertText, .alertTitle {
        color: $text-color-reverse;
      }
      .alertIcon span {
        color: rgba($text-color-reverse, 0.9) !important;
      }
    }

  }
}