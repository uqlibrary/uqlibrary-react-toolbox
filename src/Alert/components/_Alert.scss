@import '../../../sass/variables';

// Common
.alertWrapper {
  padding: 12px !important;
  .linked:hover {
    cursor: pointer;
  }
  .column{
    padding: 0px !important;
    margin: 0px !important;
  }
  .noDismiss {
    padding-right: 24px !important;
  }
  .alertIcon {
    padding: 16px !important;
    width: 40px;
    height: 40px;
    span {
      font-size: 40px !important;
    }
  }
  .alertText {
    font-weight: $text-weight-normal !important;
    font-size: $text-body-normal !important;
    margin: 12px 12px 12px 0 !important;
    overflow-wrap: break-word;
    word-wrap: break-word;
    word-break: break-word;
    -ms-hyphens: auto;
    -moz-hyphens: auto;
    -webkit-hyphens: auto;
    hyphens: auto;
    .alertTitle {
      font-weight: bold !important;
      font-size: $text-weight-normal !important
    }
  }

  // Colors and styles
  &.help {
    background-color: darken($theme-color, 5%);
    .alertText, .alertTitle {
      color: $text-color-reverse !important;
    }
    .alertIcon span {
      color: $text-color-reverse !important;
    }
    svg.alertDismiss {
      fill: $text-color-reverse !important;
    }
    button.alertAction {
      background-color: darken($theme-color, 12%) !important;
      color: $text-color-reverse !important;
    }
  }
  &.help_outline {
    background-color: lighten($secondary-color, 50%);
    .alertText, .alertTitle {
      color: $text-color !important;
    }
    .alertIcon span {
      color: $text-color !important;
    }
    svg.alertDismiss {
      fill: $text-color !important;
    }
    button.alertAction {
      background-color: lighten($secondary-color, 40%) !important;
      color: $text-color !important;
    }
  }
  &.error, &.error_outline {
    background-color: darken($error-color, 5%);
    .alertText, .alertTitle {
      color: $text-color-reverse !important;
    }
    .alertIcon span {
      color: $text-color-reverse !important;
    }
    svg.alertDismiss {
      fill: $text-color-reverse !important;
    }
    button.alertAction {
      background-color: darken($error-color, 15%) !important;
      color: $text-color-reverse !important;
    }
  }
  &.warning {
    background-color: darken($warning-color, 5%);
    .alertText, .alertTitle {
      color: $text-color-reverse !important;
    }
    .alertIcon span {
      color: $text-color-reverse !important;
    }
    svg.alertDismiss {
      fill: $text-color-reverse !important;
    }
    button.alertAction {
      background-color: darken($warning-color, 10%) !important;
      color: $text-color-reverse !important;
    }
  }
  &.info, &.info_outline {
    background-color: darken($info-color, 5%);
    .alertText, .alertTitle {
      color: $text-color-reverse;
    }
    .alertIcon span {
      color: rgba($text-color-reverse, 0.9) !important;
    }
    svg.alertDismiss {
      fill: $text-color-reverse !important;
    }
    button.alertAction {
      background-color: darken($info-color, 15%) !important;
      color: $text-color-reverse !important;
    }
  }
  &.done {
    background-color: darken($available-color, 10%);
    .alertText, .alertTitle {
      color: $text-color-reverse;
    }
    .alertIcon span {
      color: rgba($text-color-reverse, 0.9) !important;
    }
    svg.alertDismiss {
      fill: $text-color-reverse !important;
    }
    button.alertAction {
      background-color: darken($available-color, 20%) !important;
      color: $text-color-reverse !important;
    }
  }
}

// Desktop
@include tablet {
  .alertWrapper {
    margin: 12px 0;
    .alertText {
      display: flex;
      align-items: center;
    }
    button.alertAction {
      margin: 18px 0 !important;
      vertical-align: middle;
    }
    .alertDismissButton {
      margin: 12px 0 !important;
    }
  }
}

// Mobile
@include mobile {

  .alertWrapper {
    padding: 24px !important;
    margin: 6px 0 12px 0 !important;

    .alertText {
      margin: 18px 16px 12px 0 !important;
      padding-bottom: 0 !important;
    }

    button.alertAction {
      width: calc(100%);
      line-height: 18px !important;
      margin: 0 !important;
    }
  }
}
