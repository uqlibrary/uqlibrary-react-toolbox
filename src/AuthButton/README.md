# AuthButton

simple reusable login/logout button for UQ Library applications

## Props
- `isAuthorizedUser: PropTypes.bool.isRequired` - flag indicating if user is logged in or not
- `loginUrl: PropTypes.string.isRequired` - redirect URL
- `logoutUrl: PropTypes.string.isRequired` - redirect URL
- `signOutTooltipText: PropTypes.string` - default value 'Log out'
- `signInTooltipText: PropTypes.strin` - default value 'Log in'
    
## Usage

**./App.js**
```jsx
import React from 'react';
...

import {AuthButton} from 'uqlibrary-react-toolbox';

...

const App = () => (
  <div>
    ...
    <AuthButton {...props} />
    ...
  </div>
);

```

