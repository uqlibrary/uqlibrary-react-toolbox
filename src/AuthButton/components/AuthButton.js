import React from 'react';
import IconButton from 'material-ui/IconButton';
import SocialPersonOutline from 'material-ui/svg-icons/social/person-outline';
import SocialPerson from 'material-ui/svg-icons/social/person';
import PropTypes from 'prop-types';

const propTypes = {
    isAuthorizedUser: PropTypes.bool.isRequired,
    loginUrl: PropTypes.string.isRequired,
    logoutUrl: PropTypes.string.isRequired,
    signOutTooltipText: PropTypes.string,
    signInTooltipText: PropTypes.string
};

const defaultProps = {
    signOutTooltipText: 'Log out',
    signInTooltipText: 'Log in'
};

class AuthButton extends React.Component {

    redirectUser = () => {
        const redirectUrl = this.props.isAuthorizedUser ? this.props.logoutUrl : this.props.loginUrl;
        const returnUrl = window.btoa(window.location.href);
        window.location.href = `${redirectUrl}?return=${returnUrl}`;
    }

    render() {
        return (
            <div className="auth-button-wrapper">
                <IconButton tooltipPosition="bottom-left" onClick={this.redirectUser}
                    tooltip={this.props.isAuthorizedUser ? this.props.signOutTooltipText : this.props.signInTooltipText}
                    className={this.props.isAuthorizedUser ? 'log-out-button' : 'log-in-button'}>
                    {this.props.isAuthorizedUser ? <SocialPerson /> : <SocialPersonOutline />}
                </IconButton>
            </div>
        );
    }
}

AuthButton.defaultProps = defaultProps;
AuthButton.propTypes = propTypes;

export default AuthButton;
