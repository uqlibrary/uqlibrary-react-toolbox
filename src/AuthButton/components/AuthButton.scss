@import '../../../sass/variables';

.auth-button-wrapper {
  padding: 0px 10px 0 0 !important;
  margin-top: -10px;

  .auth-button-text {
    color: $text-color-reverse !important;
    font-size: 0.9em !important;
  }
}

@media (min-width: 1600px) {
  .auth-button-wrapper {
    padding: 16px 10px 0 0 !important;
    margin-top: -10px;

    .auth-button-text {
      color: $text-color-reverse !important;
      font-size: 0.75em !important;
    }
  }
}

.log-in-button, .log-out-button {
  color: $text-color-reverse;
  text-align: center;
  svg {
    fill: $text-color-reverse !important;
    color: $text-color-reverse !important;
  }
}

button.log-in-button:focus {
  background-color: #FFF !important;
  color: $theme-color-dark;
  svg {
    fill: $theme-color-dark;
    color: $theme-color-dark;
  }
}