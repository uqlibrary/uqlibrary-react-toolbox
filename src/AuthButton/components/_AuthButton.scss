@import '../../../sass/variables';

.auth-button-wrapper {
  padding: 0px 10px 0 0;
  margin-top: -10px;
}

.log-in-button, .log-out-button {
  color: $text-color-reverse;
  text-align: center;
  svg {
    fill: $text-color-reverse !important;
  }
}

button.log-in-button:focus {
  //background-color: #FFF !important;
  color: $theme-color-dark;
  svg {
    fill: $theme-color-dark;
    color: $theme-color-dark;
  }
}