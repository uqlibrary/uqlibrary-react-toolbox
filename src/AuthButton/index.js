export {default as AuthButton} from './components/AuthButton';

