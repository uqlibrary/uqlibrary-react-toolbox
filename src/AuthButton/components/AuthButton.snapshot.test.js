jest.dontMock('../components/AuthButton');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {AuthButton} from '../components/AuthButton';

function setup({...props}) {
    return shallow(<AuthButton {...props} />);
}

describe('AuthButton snapshots test', () => {
    it('renders logged out status', () => {
        const wrapper = setup({isAuthorizedUser : false});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders logged in user status', () => {
        const wrapper = setup({isAuthorizedUser : true});
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});