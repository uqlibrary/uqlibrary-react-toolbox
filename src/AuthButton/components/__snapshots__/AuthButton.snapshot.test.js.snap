// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButton snapshots test renders logged in user status 1`] = `
<div
  className="auth-button-wrapper"
>
  <IconButton
    className="log-out-button"
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    onClick={[Function]}
    tooltip="Log out"
    tooltipPosition="bottom-left"
    touch={false}
  >
    <SocialPerson />
  </IconButton>
</div>
`;

exports[`AuthButton snapshots test renders logged out status 1`] = `
<div
  className="auth-button-wrapper"
>
  <IconButton
    className="log-in-button"
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    onClick={[Function]}
    tooltip="Log in"
    tooltipPosition="bottom-left"
    touch={false}
  >
    <SocialPersonOutline />
  </IconButton>
</div>
`;
